### ISSUE TITLE
[im2txt] Early stopping

### ISSUE THREAD

**Issue by EloiZ** [tensorflow/models#773]
Hi,
is there a way to do early stopping with the im2txt model?
Imagine, we run together the training script and the evaluation script endlessly (without a fixed number of epoch). I guess we can easily keep track of the best performing checkpoint (in term of the validation loss). However, how can we kill the training thread in a clean way when the validation stopped decreasing ?
Thanks a lot,

**Comment by aselle** [comment]
This question is more appropriate for stackoverflow since it is not a bug or issue. Please re-ask it there.

**Comment by paviddavid** [comment]
@EloiZ I am struggling with the same issue. How have you solved the early stopping problem? Thanks a lot in advance!
